packages/backend/src/tools/uif/multi/MultiIndexer.ts

```diff
@@ -137,6 +141,7 @@ export abstract class MultiIndexer<T> extends ChildIndexer {
     const oldSafeHeight = (await this.getSafeHeight()) ?? safeHeight
 
     this.saved = new Map(toSave.map((c) => [c.id, c]))
+    // TODO: should we remove data in a transaction?
```

Comment:
Here the SIGTERM will not cause the bug - if the configurations are not saved (line 148) then removal will be retriggered. But the more I think about I think we should run it in transaction just for code clarity + maintainability